common/beerocks/bcl/source/son/son_wireless_utils.cpp

```diff
@@ -557,6 +559,82 @@ std::vector<uint8_t> wireless_utils::calc_5g_20MHz_subband_channels(
     return channels;
 }
 
+void wireless_utils::get_channel_preferences(
+    beerocks::message::sWifiChannel supported_channels[],
+    std::map<beerocks::message::sPreference, std::list<beerocks::message::sWifiChannel>>
+        &preferences)
+{
+    std::list<uint8_t> oper_classes;
+    wireless_utils::get_supported_operating_classes(supported_channels, oper_classes);
+
+    for (auto oper_class : oper_classes) {
+        const std::set<uint8_t> &oper_class_channels =
+            operating_classes_list.at(oper_class).channels;
+        std::list<beerocks::message::sWifiChannel> radar_affected_channels;
+        for (uint8_t i = 0; i < beerocks::message::SUPPORTED_CHANNELS_LENGTH; i++) {
+            if (oper_class_channels.find(supported_channels[i].channel) !=
+                oper_class_channels.end()) {
+                if (supported_channels[i].radar_affected)
+                    radar_affected_channels.push_back(supported_channels[i]);
+            }
```

Comment:
I think the bandwidth must be matched as well here, no?

So then, the helper function that I describe below can be used instead:

```
if (has_operating_class_channel(oper_class_channels, supported_channels[i])) {
```

Reply:
Done.